client/signup/steps/user/index.jsx

```diff
@@ -22,6 +22,19 @@ export default React.createClass( {
 		}
 	},
 
+	componentWillMount() {
```

Comment:
Essentially what this adds is a layer of caching to prevent the translation from being re-calculated on every call to `render`. That's fine, but doing it this way can leave us with output that does not change when its props do.

Ideally, React components' output (the result of `render`) should be completely dependent on their input (`this.props`). Given the same input, the output should always be the same. In this case, if the component is mounted where `this.props.subHeaderText` is "foobar" and `getFlowSteps` returns 1, `this.state.subHeaderText` will be "welcome to the wonderful wordpress.com community". If `this.props.subHeaderText` is then changed to "barfoo", then `this.state.subHeaderText` will now be "barfoo" since `componentWillMount` has already been called.

I'd suggest moving this to `componentWillReceieveProps` to handle the unlikely event that the props used in this calculation (`this.props.flowName` and `this.props.subHeaderText`) are changed after the component is mounted. Inside `componentWillReceieveProps` you could check to see if either of those props have changed, and if so, re-calculate the translation.